thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js

```diff
@@ -104,26 +104,35 @@ export default Ember.Controller.extend({
         return;
       }
 
-      entitiesService.request(context, selectedUrns);
+      // entities
+      const entitiesUrns = new Set([...selectedUrns, ...context.urns, ...context.anomalyUrns]);
+      entitiesService.request(context, entitiesUrns);
+
+      // timeseries
       timeseriesService.request(context, selectedUrns);
 
+      // breakdowns
       if (activeTab === ROOTCAUSE_TAB_DIMENSIONS) {
         const metricUrns = new Set(filterPrefix(context.urns, 'thirdeye:metric:'));
         const currentUrns = [...metricUrns].map(toCurrentUrn);
         const baselineUrns = [...metricUrns].map(toBaselineUrn);
         breakdownsService.request(context, new Set(currentUrns.concat(baselineUrns)));
       }
 
-      if (activeTab === ROOTCAUSE_TAB_METRICS) {
-        const metricUrns = new Set(filterPrefix(Object.keys(entities), 'thirdeye:metric:'));
-        const currentUrns = [...metricUrns].map(toCurrentUrn);
-        const baselineUrns = [...metricUrns].map(toBaselineUrn);
+      // aggregates
+      const offsets = ['current', 'baseline', 'wo1w', 'wo2w', 'wo3w', 'wo4w'];
+      const aggregatesUrns = new Set();
 
-        const offsets = ['wo1w', 'wo2w', 'wo3w', 'wo4w'];
-        const offsetUrns = [...metricUrns].map(urn => [].concat(offsets.map(offset => toOffsetUrn(urn, offset)))).reduce((agg, l) => agg.concat(l), []);
+      if (activeTab === ROOTCAUSE_TAB_METRICS) {
+        filterPrefix(Object.keys(entities), 'thirdeye:metric:').forEach(urn => aggregatesUrns.add(urn));
+      }
 
-        aggregatesService.request(context, new Set([...currentUrns, ...baselineUrns, ...offsetUrns]));
+      if (!_.isEmpty(context.anomalyUrns)) {
+        filterPrefix(context.anomalyUrns, 'thirdeye:metric:').forEach(urn => aggregatesUrns.add(urn));
       }
+
+      const offsetUrns = [...aggregatesUrns].map(urn => [].concat(offsets.map(offset => toOffsetUrn(urn, offset)))).reduce((agg, l) => agg.concat(l), []);
```

Comment:
It is not apparent what this is doing, let's have some comments above. Also, let's use more meaningful variable name (not just `l`)